pkg/machine/wsl/usermodenet.go

```diff
@@ -99,7 +99,7 @@ func startUserModeNetworking(mc *vmconfigs.MachineConfig) error {
 
 	// Start or reuse
 	if !running {
-		if err := launchUserModeNetDist(exe); err != nil {
+		if err := launchUserModeNetDist(exe, mc.SSH.Port); err != nil {
```

Comment:
If you are using the machine SSH port there I'm afraid there will be a conflict between gvproxy and win-sshproxy